expression/builtin_encryption_test.go

```diff
@@ -26,6 +26,45 @@ import (
 	"github.com/pingcap/tidb/util/testleak"
 )
 
+var cryptTests = []struct {
+	origin   interface{}
+	password interface{}
+	crypt    interface{}
+}{
+	{"", "", ""},
+	{"pingcap", "1234567890123456", "2C35B5A4ADF391"},
+	{"pingcap", "asdfjasfwefjfjkj", "351CC412605905"},
```

Comment:
Please add some cases for `NULL` inputs and other conditions, like:
```sql
mysql> select hex(decode("数据库", "asdfjasfwefjfjkj"));
+----------------------------------------------+
| hex(decode("数据库", "asdfjasfwefjfjkj"))    |
+----------------------------------------------+
| 8A5F1D65A3BECDD210                           |
+----------------------------------------------+
1 row in set, 1 warning (0.00 sec)

mysql> select hex(decode(12355.5555, "asdfjasfwefjfjkj"));
+---------------------------------------------+
| hex(decode(12355.5555, "asdfjasfwefjfjkj")) |
+---------------------------------------------+
| E2359202FA4CDD5AB4F2                        |
+---------------------------------------------+
1 row in set, 1 warning (0.00 sec)

mysql> select hex(decode("分布式データベース", "asdfjasfwefjfjkj"));
+----------------------------------------------------------------+
| hex(decode("分布式データベース", "asdfjasfwefjfjkj"))          |
+----------------------------------------------------------------+
| A9F2E51CEA86CC75EB4A579CA8133E9B346D5B4B610FB575637F34         |
+----------------------------------------------------------------+
1 row in set, 1 warning (0.00 sec)

mysql> select hex(decode("数据库", null));
+--------------------------------+
| hex(decode("数据库", null))    |
+--------------------------------+
| NULL                           |
+--------------------------------+
1 row in set (0.00 sec)

mysql> select hex(decode(null, "asdfjasfwefjfjkj"));
+---------------------------------------+
| hex(decode(null, "asdfjasfwefjfjkj")) |
+---------------------------------------+
| NULL                                  |
+---------------------------------------+
1 row in set (0.00 sec)

mysql> select hex(decode("pingcap", "密匙"));
+----------------------------------+
| hex(decode("pingcap", "密匙"))   |
+----------------------------------+
| CE5C02A5010010                   |
+----------------------------------+
1 row in set, 1 warning (0.01 sec)

```

Reply:
done